proof_ipa.go

```diff
@@ -83,17 +82,11 @@ type Proof struct {
 	PostValues [][]byte
 }
 
-type SuffixStateDiff struct {
-	Suffix       byte      `json:"suffix"`
-	CurrentValue *[32]byte `json:"currentValue"`
-	NewValue     *[32]byte `json:"newValue"`
-}
-
-type SuffixStateDiffs []SuffixStateDiff
-
 type StemStateDiff struct {
-	Stem        [StemSize]byte   `json:"stem"`
-	SuffixDiffs SuffixStateDiffs `json:"suffixDiffs"`
+	Stem     [StemSize]byte `json:"stem"`
+	Suffixes []byte         `json:"suffixes"`
+	Current  [][]byte       `json:"current"`
```

Comment:
We should remember that a proof verifier must now check that `[]byte` has length 32 which was implicit in the type before.